lib/interval/interval.ex

```diff
@@ -145,8 +162,12 @@ defmodule Timex.Interval do
       ["2014-09-22", "2014-09-25"]
 
   """
+  @spec with_step(Interval.t, any()) :: :ok | {:errpr, :invalid_step}
   def with_step(%__MODULE__{} = interval, step) do
-    %__MODULE__{interval | :step => step}
+    case validate_step(step) do
+      :ok -> %__MODULE__{interval | :step => step}
```

Comment:
Maybe it's worth to use `step: step` syntax here to match style of the whole file?